napari/plugins/manager.py

```diff
@@ -0,0 +1,111 @@
+import importlib
+import os
+import pkgutil
+from logging import Logger
+
+import pluggy
+
+from . import _builtins, hookspecs
+
+logger = Logger(__name__)
+
+
+class NapariPluginManager(pluggy.PluginManager):
+    PLUGIN_ENTRYPOINT = "napari.plugin"
+    PLUGIN_PREFIX = "napari_"
+
+    def __init__(self, autodiscover=True):
+        """pluggy.PluginManager subclass with napari-specific functionality
+
+        In addition to the pluggy functionality, this subclass adds
+        autodiscovery using package naming convention.
+
+        Parameters
+        ----------
+        autodiscover : bool, optional
+            Whether to autodiscover plugins by naming convention and setuptools
+            entry_points, by default True
+        """
+        super().__init__("napari")
+
+        # define hook specifications and validators
+        self.add_hookspecs(hookspecs)
+
+        # register our own built plugins
+        self.register(_builtins, name='builtins')
+        # discover external plugins
+        if not os.environ.get("NAPARI_DISABLE_PLUGIN_AUTOLOAD"):
+            if autodiscover:
+                self.discover()
+
+    def discover(self):
+        """Discover modules by both naming convention and entry_points
+
+        1) Using naming convention:
+            plugins installed in the environment that follow a naming
+            convention (e.g. "napari_plugin"), can be discovered using
+            `pkgutil`. This also enables easy discovery on pypi
+
+        2) Using package metadata:
+            plugins that declare a special key (self.PLUGIN_ENTRYPOINT) in
+            their setup.py `entry_points`.  discovered using `pkg_resources`.
+
+        https://packaging.python.org/guides/creating-and-discovering-plugins/
+
+        Returns
+        -------
+        int
+            The number of modules successfully loaded.
+        """
+        count = 0
+        if not os.environ.get("NAPARI_DISABLE_ENTRYPOINT_PLUGINS"):
+            # register modules defining the napari entry_point in setup.py
+            count += self.load_setuptools_entrypoints(self.PLUGIN_ENTRYPOINT)
+        if not os.environ.get("NAPARI_DISABLE_NAMEPREFIX_PLUGINS"):
+            # register modules using naming convention
+            count += self.load_modules_by_prefix(self.PLUGIN_PREFIX)
+
+        if count:
+            msg = f'loaded {count} plugins:\n  '
+            msg += "\n  ".join([n for n, m in self.list_name_plugin()])
+            logger.info(msg)
```

Comment:
the logs are written to stdout by default (unless we configure a handler to write to file, which we could do), but the default logging level for the basicConfig is "warning" ... so in order to see these logs, you would have to run the following code _before_ the manager plugin gets imported.

```python
import logging
logging.basicConfig(level=logging.INFO)
```
that's a bit tricky at the moment, since our `__init__.py` imports so much of the library before __main__ gets to run, (so it would be hard for us to add a `--loglevel` flag to the CLI, for the same reason [it was hard to make an `--info` flag](https://github.com/napari/napari/pull/814#issuecomment-567628580) that worked without failing on import).  But, that's sort of a separate issue; I really think we're going to need some logging at some point in order to help people debug stuff.  So here, until we add something like loguru, I don't think it hurts to use stdlib (and will make it easier to grep logging "hotspots" if we move to something like that).

Reply:
An env variable could work here. (can be in a later PR)